### PR TITLE
#926 Practitioner input

### PR DESCRIPTION
**Pull request by CarlosNZ**
Just putting this in Draft for now cos there's a couple of things not working properly (doesn't currently compile in front-end).

@clemens-msupply Could use your help figuring out how to add the "clinician" property to the "Encounter" type. You'll notice in "CreateEncounterModal" that there is an error with `draft?.clinician` as property is not defined in "EncounterFragment", but this comes from database types as far as I could tell.

Also -- I'm currently just saving the whole "clinician" object in the Encounter document, but should it also be at the "root" level of the Encounter data?

Currently this is very basic -- it just displays a "clinician" drop-down on the Encounter create modal, and then displays that value on the Encounter details page. Should this be editable on the Details page? (It's currently displayed as a disabled input like the Patient name is)
Screenshot 2023-02-02 at 11 38 07 AM
Screenshot 2023-02-02 at 11 38 20 AM

### CODE REVIEW

File: client/packages/common/src/intl/locales/en/patients.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "label.visit-date": "Visit Date",
   "label.visit-start": "Start",
   "label.visit-end": "End",
+  "label.seen-by": "Seen by",
```
Comment (CarlosNZ) [inline client/packages/common/src/intl/locales/en/patients.json:21]: "Seen by"? or should it just be "Clinician"?

File: server/service/src/sync/program_schemas/hiv_care_program.json [hunk 2]
```diff
@@ -107,8 +107,8 @@
         "mother": {
           "$ref": "#/definitions/HIVRelation"
         },
-        "partnerHivStatus": {
-          "$ref": "#/definitions/HIVPartnerStatus"
+        "partnerHIVStatus": {
```
Comment (CarlosNZ) [inline server/service/src/sync/program_schemas/hiv_care_program.json:110]: Unrelated, but needed to be changed to match the change @clemens-msupply did yesterday (I think)

File: server/service/src/sync/program_schemas/hiv_care_program.json [hunk 3]
```diff
@@ -246,7 +246,7 @@
           "$ref": "#/definitions/SocioEconomics"
         }
       },
-      "required": ["isDeceased", "contactDetails", "socioEconomics"],
+      "required": ["contactDetails"],
```
Comment (CarlosNZ) [inline server/service/src/sync/program_schemas/hiv_care_program.json:249]: Also changed so Enrolment form would validate

File: client/packages/common/src/intl/locales/en/patients.json (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/common/src/intl/locales/en/patients.json:21]: good question @richardmoizeau  ? "Clinician" sound clearer?

Comment (richardmoizeau) [reply client/packages/common/src/intl/locales/en/patients.json:21]: Yes, `Clinician` is better than `Seen by`

**Comment by clemens-msupply** [comment]
Yes its a bit tricky and can do with some clearer types, currently the Encounter type is defined as:

```typescript
type Encounter = Pick<
  EncounterFragment,
  'startDatetime' | 'endDatetime' | 'status'
>;
```

This type doesn't has much do to with EncounterFragment from the encounter api. It is just supposed to be the generic document type for an encounter (the one that is defined in the json schema).

so you can just copy the type from the schema repo, e.g.:

```typescript
export interface Encounter {
  status?: EncounterStatus;

  /**
   * Encounter start date and time
   *
   * @format date-time
   */
  startDatetime: string;
  /**
   * Encounter end date and time
   *
   * @format date-time
   */
  endDatetime?: string;

  practitioner?: Practitioner;

  //type?: VisitType;
}
```
> Also -- I'm currently just saving the whole "clinician" object in the Encounter document, but should it also be at the "root" level of the Encounter data?

yes this would be useful, e.g. to search for encounters for a certain practitioner will write an issue for it!

I think this would also makes it easier to access this information in the header because you don't need to look into the raw document data?

File: client/packages/common/src/intl/locales/en/patients.json (hunk 1 above)
Comment (CarlosNZ) [reply client/packages/common/src/intl/locales/en/patients.json]: 👍 "Clinician" is is! f191ed3e7

**Comment by CarlosNZ** [comment]
Changed to non-Draft now, as I've fixed the types (in that they compile). I'll do an additional PR where the clinician info will get written to the Encounter root (as mentioned above) based of #1129 .

**Review by clemens-msupply — CHANGES_REQUESTED** [review]
thanks looks very good, I discovered one crash for an edge case where clinicians don't have a first name. The api call fails with:

```json
{
    "data": null,
    "errors": [
        {
            "message": "Bad user input",
            "locations": [
                {
                    "line": 2,
                    "column": 3
                }
            ],
            "path": [
                "insertEncounter"
            ],
            "extensions": {
                "details": "InvalidDataSchema(\n    [\n        \"null is not of type \\\"string\\\"\",\n    ],\n)"
            }
        }
    ]
}
```

I think you need to clean up the data you are sending there are nulls and additional data in it for some reason...

image

There are some other minor related issues, see comments

**Comment by CarlosNZ** [comment]
> I think you need to clean up the data you are sending there are nulls and additional data in it for some reason...

Yeah sure, I've stripped it down to a simplified "Clinician" (firstName, lastName, id).

Also, the name fields become empty string if `null` in database.

Including other suggested fixes: c86eb9c7e

Cheers.

**Review by clemens-msupply — APPROVED** [review]
Works great, thanks!

**Pull request merged**